ionoscloud/provider.go

```diff
@@ -71,7 +71,7 @@ func Provider() *schema.Provider {
 			ServerResource:             resourceServer(),
 			VolumeResource:             resourceVolume(),
 			GroupResource:              resourceGroup(),
-			"ionoscloud_share":         resourceShare(),
+			shareResource:              resourceShare(),
```

Comment:
I think we should be consistent regarding naming - ShareResource instead of shareResource

Reply:
That would mean exporting the constant to other packages. The question is, do we want to do that?

Reply:
Will change to uppercase, maybe we will need these in other packages